iac-extensions/arm/src/main/java/org/sonar/iac/arm/visitors/ArmSymbolVisitor.java

```diff
@@ -77,24 +83,24 @@ void visitFile(File file) {
     file.setSymbolTable(currentSymbolTable);
   }
 
-  private void visitVariableDeclaration(VariableDeclaration variableDeclaration) {
-    var symbol = currentSymbolTable.addSymbol(variableDeclaration.declaratedName().value());
-    symbol.addUsage(variableDeclaration, Usage.Kind.ASSIGNMENT);
+  private void visitDeclaration(Declaration declaration) {
+    var symbol = currentSymbolTable.addSymbol(declaration.declaratedName().value());
+    symbol.addUsage(declaration, Usage.Kind.ASSIGNMENT);
   }
 
-  void visitVariable(Variable variable) {
-    if (variable.symbol() != null) {
+  void visitHasIdentifier(HasIdentifier hasIdentifier) {
```

Comment:
Maybe just `tree` will be a better name instead of `hasIdentifier`? Now the name looks like a boolean flag.